Master branch broken on M1 Mac

Getting a flood of these errors:
```text
error: brackets are not allowed here; to declare an array, place the brackets after the identifier
In file included from opal_datatype_pack.c:40:
In file included from ../../opal/datatype/opal_datatype_pack.h:23:
../../opal/datatype/opal_datatype_pack_unpack_predefined.h:370:13: error: expected ')'
            OPAL_DATATYPE_MYUNPACK(SHORT_FLOAT_COMPLEX);
            ^
../../opal/datatype/opal_datatype_pack_unpack_predefined.h:304:13: note: expanded from macro 'OPAL_DATATYPE_MYUNPACK'
            OPAL_DATATYPE_MYUNPACK_AVAILABLE,      \
            ^
../../opal/datatype/opal_datatype_pack_unpack_predefined.h:370:13: note: to match this '('
../../opal/datatype/opal_datatype_pack_unpack_predefined.h:304:13: note: expanded from macro 'OPAL_DATATYPE_MYUNPACK'
            OPAL_DATATYPE_MYUNPACK_AVAILABLE,      \
            ^
```
Looks like something in the datatype engine is broken?

--- a/opal/include/opal_config_bottom.h
+++ b/opal/include/opal_config_bottom.h
@@ -522,6 +522,14 @@ static inline uint16_t ntohs(uint16_t netvar) { return netvar; }
 #define restrict
 #endif
 
+#ifdef HAVE_OPAL_SHORT_FLOAT_COMPLEX_T
+#undef opal_short_float_complex_t
+typedef struct {
+    opal_short_float_t real;
+    opal_short_float_t imag;
+} opal_short_float_complex_t;
+#endif
+
 #else
 
 /* For a similar reason to what is listed in opal_config_top.h, we

